frontend/src/app/components/workflow-editor/workflow-editor.component.ts

```diff
@@ -261,31 +259,65 @@ export class WorkflowEditorComponent {
     if (element.getAttribute('dispatcher') !== 'link') {
       return;
     }
-    const link = this.currentWorkflow.links.find(
+    const link = this.currentWorkflow.content.links.find(
       wfLink => wfLink.id === element.id
     );
     if (link !== undefined) {
       return;
     }
-    const linkSourceIds = this.flowchartConverter.getLinkOperatorAndConnector(
+
+    const linkSourceIds: VertexIds = this.flowchartConverter.getLinkOperatorAndConnectorId(
       element,
       true
     );
-    const linkTargetIds = this.flowchartConverter.getLinkOperatorAndConnector(
+    const linkTargetIds: VertexIds = this.flowchartConverter.getLinkOperatorAndConnectorId(
       element,
       false
     );
-    const linkPath = this.flowchartConverter.convertLinkPathToPoints(element);
 
-    const newLink: WorkflowLink = {
+    let startConnector: Connector;
+    let endConnector: Connector;
+
+    if (linkSourceIds.operatorId === this.currentWorkflow.id) {
```

Comment:
const sourceIsWorkflowInput = linkSourceIds.operatorId === this.currentWorkflow.id
const startConnector: Connector = this.flowchartConverter.getConnectorFromOperatorById(
        linkSourceIds,
        this.currentWorkflow,
        sourceIsWorkflowInput
      );
const targetIsWorkflowOutput = linkTargetIds.operatorId === this.currentWorkflow.id;
const endConnector: Connector = this.flowchartConverter.getConnectorFromOperatorById(
        linkTargetIds,
        this.currentWorkflow,
        targetIsWorkflowOutput
      );